test/parallel/test-icu-transcode.js

```diff
@@ -39,6 +39,10 @@ for (const test in tests) {
 }
 
 assert.throws(
+  () => buffer.transcode(null, 'utf8', 'ascii'),
+  /TypeError: "source" argument must be a Buffer/
```

Comment:
If you're testing the entire error message, you can add `^` and `$` to the regular expression.